source/parser.hera

```diff
@@ -1073,29 +1073,55 @@ FunctionExpression
     }
 
   # Ruby/Crystal style block shorthand
-  Ampersand !_ CallExpressionRest+ ->
-    $1.token = "$ => $"
-    return {
-      type: "ArrowFunction",
-      children: [$1, $3],
+  AmpersandUnaryPrefix?:prefix Ampersand AmpersandBlockRHS?:rhs ->
+    if (!prefix && !rhs) return $skip
```

Comment:
Removing that check causes a bit of havoc with binary `&` and `&&`:

Ex:
```
--- Source   ---
a && b

--- Expected ---
a && b

--- Got      ---
a($ => $& b)
```

Reply:
It may be possible to add some lookahead assertions so we could eventually get `&` as an identity function shorthand (a valuable goal) but probably not for this PR.